src/main/java/oracle/r2dbc/impl/OracleConnectionImpl.java

```diff
@@ -672,18 +714,29 @@ public IsolationLevel getTransactionIsolationLevel() {
     requireNonNull(isolationLevel, "isolationLevel is null");
     requireOpenConnection(jdbcConnection);
 
-    // TODO: Need to add a connection factory option that disables Oracle
-    //  JDBC's Result Set caching function before SERIALIZABLE can be supported.
-    // For now, the isolation level can never be changed from the default READ
-    // COMMITTED.
-    if (isolationLevel.equals(READ_COMMITTED)) {
+    // Do nothing if the level isn't changed
+    if (isolationLevel.equals(this.isolationLevel))
       return Mono.empty();
+
+    // Compose a command to set the isolation level of the database session:
+    // ALTER SESSION SET ISOLATION_LEVEL = {SERIALIZABLE | READ COMMITTED}
+    String alterSession = "ALTER SESSION SET ISOLATION_LEVEL = ";
```

Comment:
Why not use the JDBC API to change the isolation?
http://ojdbc.us.oracle.com/JDBC_Javadoc/MAIN/latest/java/sql/Connection.html#setTransactionIsolation(int)

Reply:
Oracle JDBC implements setTransactionIsolation with a blocking database call. It will execute a ALTER SESSION SET ISOLATION_LEVEL with java.sql.Statement.execute(). For R2DBC, we can use io.r2dbc.spi.Statement.execute() to execute the same command as a non-blocking call.

When we by-pass the JDBC API, it means that Oracle JDBC longer retains the correction isolation level as a field of java.sql.Connection. This will effect the return value of java.sql.Connection.getTransactionIsolation(), but it doesn't effect any other behavior of Oracle JDBC.

This branch has Oracle R2DBC retain the correct isolation level as a field of oracle.r2dbc.impl.OracleConnectionImpl, and return that value when getTransactionIsolationLevel is called.